infinitered/ignite-ir-boilerplate
# Introduce Storybook into Ignite

In this PR I introduce Storybook and Storyshots to the core of the boilerplate and setup two basic stories against some components that ship with the boilerplate. I am adding this as I believe Storybook and Storyshots to be fantastic tools when building out a set of components like many end users of this boilerplate will be doing. This is not meant to be complete but rather a glimpse into what this type of integration could look like. If it is liked I can finish out writing the component stories so all components ship with storybook stories.

# What was added
Two new dependencies for Storybook and Storyshots, a storybook script, a few stories for existing components in the boilerplate, a stories test to run storyshots and the support storybook folder to help run storybook

# What needs to be done
- Write more stories
- Possibly clean up the implementation into the boilerplate
- Docs around storybook and storyshots
- Make sure storyshots runs on precommit or prepush hooks to ensure snapshots did not change

# What does this look like?
This is what Storybook looks like:
![screen shot 2017-08-03 at 7 37 56 pm](https://user-images.githubusercontent.com/14151327/28948376-480ed850-7883-11e7-9443-fa1576199498.png)

For storyshots I will note pollute you with a snapshot file so I will just say that it generates your typical Jest snapshot for each and every story defined.

# How do you test it?
Storybook: Instead of running `(npm/yarn) start` in the project directory you run `(npm/yarn) run storybook` then launch the app normally (via `react-native run-ios` and `react-native run-android`). ANDROID NOTE: You will need to run `adb reverse tcp:7007 tcp:7007` before launching the app to ensure port 7007 is tunneled back to your computer

Storyshots: `npm test` will run the snapshot tests setup for stories.
